include/materials/material_utility.tcc

```diff
@@ -203,3 +203,29 @@ inline const Eigen::Matrix<double, 6, 1> mpm::materials::dtheta_dsigma(
   // Compute and return dtheta / dsigma
   return (dtheta_dr * ((dr_dj2 * dj2_dsigma) + (dr_dj3 * dj3_dsigma)));
 }
+
+//! Compute incremental dplastic_strain
+inline const Eigen::Matrix<double, 6, 1> mpm::materials::dplastic_strain(
```

Comment:
I think this should not be here as different methods have different ways of updating plastic strain. Return mapping algorithm append plastic strain incrementally for example.

Reply:
But this is correct no matter what model you use. You can choose to use it or not

Reply:
That's true. I am okay with it, but I think we should just put this on the law. Passing all the tensors and matrices to a function and only to perform an inversion is not that efficient I guess...

Reply:
I agree with @bodhinandach this might be easier to do locally in the model

Reply:
But this should always be correct right, since even for different methods, dstrain is always divided into dstrain_elastic and dstrain_plastic anyways. The formulation of D stiffness matrix is defined in such a way that we subtract D_elastic. So this is correct.

It's a hassle though, for one line of code, we have to transfer a lot of variables

Reply:
Yes, exactly, I am not saying that this is wrong. I just feel that this shouldn't be here due to inefficiency and it is easier that each model can perform their own way of computing `dplastic_strain`.

Reply:
How about we have this option, and each material can choose whether to use this or not. I do feel it is inefficient but so does passing `stress` to only compute `p` right?

Reply:
OK! Waiting for your approval @bodhinandach

Reply:
I am still not sure that this is the best implementation practice. My concern is related to points made by @kks32 and some other points. 

As @kks32 mentioned, this passes 48 elements, 8 times more to compute p, so the scale is a bit different. Imagine if you call this for 1b billion particles.

As for `p`, `j2`, and `j3` they are universal and can be used for all types of stresses. However, this specific function only works and correct for models with small-strain elastoplastic assumptions, most likely with explicit implementation. The kinematic decomposition there won't be quite useful for models with large deformation (they use multiplicative decomp), materials with viscous effect (viscous terms are missing), or materials with constrained strains. And therefore, bring back to my earlier point, why don't we just implement this inside each model?

In conclusion, I can approve this, but this won't give any benefits as compared to computing invariants. There are many limitations in both efficiency and usefulness. I'll let the group decide.

Reply:
A few thoughts:
- I agree this won't work for viscosity, but I think it should work for any elastoplastic materials which are the materials that we usually use anyways. I think it should work for large strains too, and it will give the same results with using `lambda` for instance.
- I agree it's not that efficient with 48 doubles to get one single double. However, coming back to comparison with the computation of p, in NorSand at least, one step of compute stress would require at least 4 calls to compute p, which might be more efficient if we do them within the material model itself.
- My point being, not all functions in `mpm::materials` would be used by all material models. We can choose whether to use this function or not and that is fine because we are doing inline anyways. Didn't we want to make as uniform definitions so it is easier to maintain?

I will wait till tomorrow's discussion before merging this PR.